tools/tools.mk

```diff
@@ -1,7 +1,33 @@
 noinst_PROGRAMS += tools/tableidxgen tools/policyexamplegen
 
+TOOLS_CFLAGS = -I${srcdir}/usr/include -I${srcdir}/usr/lib/api -I${top_builddir}/usr/lib/api
+
 tools_tableidxgen_SOURCES = tools/tableidxgen.c usr/lib/api/mechtable.inc
-tools_tableidxgen_CFLAGS = -I${srcdir}/usr/include -I${srcdir}/usr/lib/api
 
 tools_policyexamplegen_SOURCES = tools/policyexamplegen.c usr/lib/api/mechtable.c
-tools_policyexamplegen_CFLAGS = -I${srcdir}/usr/include -I${srcdir}/usr/lib/api -I${top_builddir}/usr/lib/api
+
+if CROSS
+tools_tableidxgen_LDADD =
+tools_tableidxgen_LINK = $(LIBTOOL) --tag=CC --mode=link $(CC_FOR_BUILD) \
+		$(CFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) -o $@
+
+$(tools_tableidxgen_OBJECTS): CC=$(CC_FOR_BUILD)
+$(tools_tableidxgen_OBJECTS): CFLAGS=$(CFLAGS_FOR_BUILD) $(TOOLS_CFLAGS)
+$(tools_tableidxgen_OBJECTS): LDFLAGS=$(LDFLAGS_FOR_BUILD)
+
+tools_policyexamplegen_LDADD =
+tools_policyexamplegen_LINK = $(LIBTOOL) --tag=CC --mode=link $(CC_FOR_BUILD) \
```

Comment:
Any particular reason why we use libtool to build the policy generator?  It should not have any dependencies to a library that requires libtool.

Reply:
no, no reason. I just copied that from somewhere else :-) 
What would you suggest instead? The point is that we need to use $(CC_FOR_BUILD) for building this.

Reply:
How about simply
```
$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) -o $@
```